### DISCUSSION TITLE
A rather confusing behavior in Lua

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#223] [General]
Both Lua and Nelua return a `bash: !': event not found` when you attempt to run `lua -e "print('hello!')"` and `nelua -i "print('hello!')"`, respectively; but when you enter in Lua interactive mode this works! :confused: 

![image](https://github.com/edubart/nelua-lang/assets/1777466/373316f8-84c0-4046-946d-b209be63102b)

@edubart since you are more experienced in Lua, can you enlighten me please why this peculiar behavior happens in the first place?

**Comment by edubart** [comment]
Actually this an issue with `bash` shell, when using double quotes with bash commands should be very careful about its contents, `!` has specially meaning for bash. The fix is to use `lua -e 'print("hello!")'` and `nelua -i 'print("hello!")'`.

**Reply by stefanos82** [reply]
Confirmed...darn it bash, why is it always you behind every trouble!

![image](https://github.com/edubart/nelua-lang/assets/1777466/fa214f24-cec5-43dc-9d54-10cd3c332227)

**Reply by RaycatWhoDat** [reply]
As an aside, the first result of a Google search also gives you this answer. No need to wait for a discussion response.

**Reply by stefanos82** [reply]
I guess the results differ from place to place, thanks to geolocation.

The results I got were not clear enough and preferred to ask Eduardo that more or less specializes in Lua; plus I chose the "General" category here in Discussions for this particular reason, for generic discussion; that's all :man_shrugging:
